Samples/NetworkMediaStreamer/RTSPServer/src/SocketWrapper.cpp

```diff
@@ -0,0 +1,304 @@
+                bFirstHandshake ? NULL : &pSock->m_hCtxt,
+                &InBuffDesc,
+                Attribs,
+                SECURITY_NETWORK_DREP,
+                bFirstHandshake ? &pSock->m_hCtxt : NULL,
+                &OutBuffDesc,
+                &Attribs,
+                &tokenLifetime);
+
+            if (InSecBuff[1].BufferType == SECBUFFER_EXTRA)
+            {
+                //TODO: handle this case
+            }
+
+            winrt::check_hresult(pSock->m_SecurityStatus);
+            send(pSock->m_socket, (char*)OutSecBuff.pvBuffer, OutSecBuff.cbBuffer, 0);
+        }
+
+    }
+    catch(...)
+    {
+        pSock->m_SecurityStatus = winrt::to_hresult();
+        pSock->m_hCtxt.dwUpper = pSock->m_hCtxt.dwLower = 0;
+    }
+
+    if (((SEC_I_CONTINUE_NEEDED == pSock->m_SecurityStatus)
+        || (SEC_I_COMPLETE_AND_CONTINUE == pSock->m_SecurityStatus)))
+    {
+        UnregisterWait(pSock->m_callBackHandle.detach());
+        RegisterWaitForSingleObject(pSock->m_callBackHandle.put(), pSock->m_readEvent.get(), (WAITORTIMERCALLBACK)ReadDelegate, pSock, INFINITE, WT_EXECUTEONLYONCE);
+    }
+    else
+    {
+        pSock->m_handshakeDone.notify_all();
+    }
+
+}
+
+void CSocketWrapper::InitilizeSecurity()
+{
+    TimeStamp         Lifetime;
+    PSecPkgInfo pPkgInfo = nullptr;
+    winrt::check_hresult(QuerySecurityPackageInfo((LPWSTR)g_lpPackageName, &pPkgInfo));
+
+    auto maxTokenSz = pPkgInfo->cbMaxToken;
+    m_bufSz = pPkgInfo->cbMaxToken;
+    m_pInBuf = std::make_unique<BYTE[]>(pPkgInfo->cbMaxToken);
+    m_pOutBuf = std::make_unique<BYTE[]>(pPkgInfo->cbMaxToken);
+    FreeContextBuffer(pPkgInfo);
+
+    // Perform Handshake
+    SCH_CREDENTIALS credData;
+    ZeroMemory(&credData, sizeof(credData));
+    credData.dwVersion = SCH_CREDENTIALS_VERSION;
+    credData.dwCredFormat = SCH_CRED_FORMAT_CERT_HASH;
+    credData.cCreds = (DWORD)m_aCertContext.size();
+    credData.paCred = m_aCertContext.data();
+
+    winrt::check_hresult(AcquireCredentialsHandle(
+        NULL,
+        g_lpPackageName,
+        SECPKG_CRED_INBOUND,
+        NULL,
+        &credData,
+        NULL,
+        NULL,
+        &m_hCred,
+        &Lifetime));
+
+    m_readEvent.attach(WSACreateEvent());      // create READ wait event for our RTSP client socket
+    if (!m_readEvent) // == WSA_INVALID_EVENT)
+    {
+        winrt::check_win32(WSAGetLastError());
+    }
+
+    std::mutex mutexDone;
+    std::unique_lock<std::mutex> lockDone(mutexDone);
+
+    WSAEventSelect(m_socket, m_readEvent.get(), FD_READ | FD_CLOSE);   // select socket read event
+    RegisterWaitForSingleObject(m_callBackHandle.put(), m_readEvent.get(), (WAITORTIMERCALLBACK)ReadDelegate, this, INFINITE, WT_EXECUTEONLYONCE);
+
+    m_handshakeDone.wait(lockDone);
+    UnregisterWait(m_callBackHandle.detach());
+    WSACloseEvent(m_readEvent.detach());
+
+    winrt::check_hresult(m_SecurityStatus);
+}
+
+int  CSocketWrapper::Recv(BYTE* buf, int sz)

```

Comment:
>   [](start = 3, length = 2)

a couple places with weird double or more whitespaces, you could ctrl+A + ctrl+K + ctrl+F in VS to auto indent everything in all your files